### PR TITLE
Fix compiling on Windows systems with Japanese code pages (issue 6404) for MSVC 2015/2017

### PR DESCRIPTION
**Pull request by Dwedit**
## Description

Per discussion on Issue #6404, there are build issues if your compiler is MSVC 2015 or 2017, and you are on a system with Japanese system locale (code page 932).

The resolution to the problem is to remove #pragma execution_character_set("utf-8") on MSVC 2015 and later, and instead specify the /utf-8 command line switch.

This block in several files:
```
#if defined(_MSC_VER) && !defined(_XBOX)
/* https://support.microsoft.com/en-us/kb/980263 */
#pragma execution_character_set("utf-8")
#endif
```

Is getting changed to this:
```
#if (_MSC_VER >= 1500 && _MSC_VER < 1900) && !defined(_XBOX)
/* https://support.microsoft.com/en-us/kb/980263 */
#pragma execution_character_set("utf-8")
#endif
```

I also added the /bigobj compiler switch to the MSVC2017 debug builds because it won't build without that switch present.

I also **blindly** added the /bigobj switch to the MSVC2015 debug builds as well, since I do not have that compiler installed, and do not know if the debug builds will work without that switch.  The compilers seem similar enough that it might be necessary.

Also re-added UTF-8 BOM markers once again...  If you see the first line changed to the same thing, that's a BOM marker addition.

## Related Issues

Issue #6404

### CODE REVIEW

**Comment by aliaspider** [comment]
HAVE_LANG_EXTRA isn't defined for travis builds, otherwise it would have detected that this change will not build on most targets.

File: intl/msg_hash_ar.c [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #include "../configuration.h"
 #include "../verbosity.h"
 
-#if defined(_MSC_VER) && !defined(_XBOX)
+#if (_MSC_VER >= 1500 && _MSC_VER < 1900) && !defined(_XBOX)
```
Comment (aliaspider) [inline intl/msg_hash_ar.c]: you still need to check if _MSC_VER is defined first.
also, is the upper bound necessary ?

Comment (Dwedit) [reply intl/msg_hash_ar.c]: `#pragma execution_character_set("utf-8")` is a hack that Microsoft introduced in MSVC 2008, or `_MSC_VER >= 1500`.  It was deprecated as of MSVC 2015, or `_MSC_VER < 1900`, and causes problems if combined with the /utf-8 switch which replaced it.

**Comment by twinaphex** [comment]
@bparker06 Will this still compile right with earlier MSVC versions?

**Comment by Dwedit** [comment]
Regarding Pre-2008 MSVC and UTF-8 support...
I think it will compile properly if you're on a SBCS code page (such as Western-1252), and fail if you're on a MBCS code page (such as Shift-JIS 932).  Shift-JIS is always problematic since it includes things like slashes and quotation marks as the second byte of a character.
I do not know the exact version of MSVC that added support for BOM markers, they may need to be removed to build it on a much earlier version of MSVC.  Haven't yet tried going all the way down to MSVC 6.0.

**Comment by bparker06** [comment]
I believe we disable HAVE_LANGEXTRA for pre-2010 versions, and that works even down to VC6.

**Comment by twinaphex** [comment]
Hi @bparker06 , what is the status on this PR? Is this safe to merge or not? Are there backwards compatibility issues with earlier MSVC versions that still need to be resolved?

**Pull request closed without merging**